p2pd/main.go

```diff
@@ -1,28 +1,128 @@
 package main
 
+import "C"
 import (
 	"context"
 	"flag"
 	"fmt"
+	"io"
 	"log"
+	"os"
 	"strings"
 
 	libp2p "github.com/libp2p/go-libp2p"
 	connmgr "github.com/libp2p/go-libp2p-connmgr"
-	p2pd "github.com/libp2p/go-libp2p-daemon"
+	peer "github.com/libp2p/go-libp2p-peer"
 	identify "github.com/libp2p/go-libp2p/p2p/protocol/identify"
+	multiaddr "github.com/multiformats/go-multiaddr"
+
+	p2pd "github.com/libp2p/go-libp2p-daemon"
+	"github.com/libp2p/go-libp2p-daemon/p2pclient/go"
 )
 
+// ClientConfig defines the configuration options for the client
+type ClientConfig struct {
+	pathd   string
+	pathc   string
+	command string
+	args    []string
+}
+
 func main() {
+	//defined interactions between client and daemon
+	commands := [...]string{"Identify", "Connect"}
+	client := flag.Bool("client", false, "run in client mode")
+	pathd := flag.String("pathd", "/tmp/p2pd.sock", "daemon control socket path")
+	pathc := flag.String("pathc", "/tmp/p2pc.sock", "client control socket path")
+	command := flag.String("command", commands[0], "command to send to the daemon")
+	flag.Parse()
+
+	if *client {
+		config := ClientConfig{
+			pathd:   *pathd,
+			pathc:   *pathc,
+			command: *command,
+			args:    flag.Args(),
+		}
+		startC(config)
+	} else {
+		startD()
+	}
+}
+func startC(config ClientConfig) {
```

Comment:
Please make a separate binary for the client; you can make a directory in the top level (call it `p2pc`) and add a main program in there.

Reply:
> Please make a separate binary for the client; you can make a directory in the top level (call it p2pc) and add a main program in there.

Will do...it was bothering me too :)